### ISSUE TITLE
sd-netlink: callback failed: Invalid argument

### ISSUE THREAD

**Issue by taneli76** [systemd/systemd#10320]
**systemd version the issue has been seen with**
> systemd-239 as packaged in ubuntu cosmic systemd-239-7ubuntu9

**Used distribution**
 > ubuntu 18.10 cosmic Beta as of Oct 9

**Expected behaviour you didn't see**
 > Assigned IPv4 and IPv6 addresses

**Unexpected behaviour you saw**
 > No addresses assigned, but:
```    PRIORITY=7
    SYSLOG_FACILITY=3
    _SELINUX_CONTEXT=unconfined
    _SYSTEMD_SLICE=system.slice
    _TRANSPORT=journal
    SYSLOG_IDENTIFIER=systemd-networkd
    _UID=100
    _GID=102
    _COMM=systemd-network
    _EXE=/lib/systemd/systemd-networkd
    _CMDLINE=/lib/systemd/systemd-networkd
    _CAP_EFFECTIVE=3c00
    _SYSTEMD_CGROUP=/system.slice/systemd-networkd.service
    _SYSTEMD_UNIT=systemd-networkd.service
    _PID=736
    CODE_FILE=../src/libsystemd/sd-netlink/sd-netlink.c
    CODE_FUNC=process_reply
    ERRNO=22
    MESSAGE=sd-netlink: callback failed: Invalid argument
    _BOOT_ID=555a32bf752f4963af66b9d6888abcae
    _SYSTEMD_INVOCATION_ID=a8d0b8fe61ce454487a945c975bcb336
    CODE_LINE=348
    _SOURCE_REALTIME_TIMESTAMP=1539028545269419
```

**Steps to reproduce the problem**

Attempt to configure IPv4 using DHCP and IPv6 through DHCPv6-PD and static address, in particular, `/etc/netplan/01-netcfg.yml`:

```network:
  version: 2
  renderer: networkd
  ethernets:
    enp0s20:
      dhcp4: yes
      dhcp6: no
      address: ["2001:CAFE:BABE:1000::1000/128"]
```
(variants tried: `dhcp6: yes`, and without `dhcp6` and `addresses` fields, at least).

And `/etc/systemd/networkd/10-enp0s20.cfg`, roughly so (there were numerous variants of this over the course of two days):
```[Match]
Interface=enp0s20

[Network]
DHCP=ipv6  # also yes, dhcpv6
IPv6AcceptRA=true  # also tried false
IPv6PrefixDelegation=dhcpv6  # the host gets IPv6 by prefix delegation

[DHCP]
# This was mentioned somewhere to help with a seemingly related bug, but did not
#Hostname=sd-123456
DUIDType=vendor
DUIDRawData=01:02:03:04:05:06:07:08:09:10  # used the actual DUID from provider, which has 10 hex digits

[IPv6PrefixDelegation]  # also tried without this whole section
Managed=true
#Prefix=2001:CAFE:BABE:1000::1000/56  # tried also this at some point, in desperation
```

Moreover, dhclient works ok with `dhclient -cf /etc/dhcp/dhclient6.conf -6 -P -v enp0s20` after placing `interface "eth0" { send dhcp6.client-id DUID; request; }` in its configuration file.

**Comment by yuwata** [comment]
Could you provide relevant debug logs, that is, run networkd with `SYSTEMD_LOG_LEVEL=debug` and provide journal logs.
Also, please show minimal .network config file which reproduces the issue. Is it enough to set only `DUIDType=vendor` and `DUIDRawData=` for triggering the issue?
Moreover, if possible, please also test with the recent snapshot and #10201.

**Comment by taneli76** [comment]
Dear @yuwata I no longer have access to the system. Before it was wiped out, I collected the following logs (I've redacted all hex strings from it, since I did not know which ones might be sensitive):
```
Mon 2018-10-08 21:55:45.269419 CEST [<redacted>]
    _MACHINE_ID=<redacted>
    _HOSTNAME=sd-123456
    PRIORITY=7
    SYSLOG_FACILITY=3                                                                                                                                                                          
    _EXE=/lib/systemd/systemd-networkd                                                                                                                                                         
    _CMDLINE=/lib/systemd/systemd-networkd                                                                                                                                                     
    _CAP_EFFECTIVE=3c00                                                                                                                                                                        
    _SYSTEMD_CGROUP=/system.slice/systemd-networkd.service                                                                                                                                     
    _SYSTEMD_UNIT=systemd-networkd.service                                                                                                                                                     
    _PID=736                                                                                                                                                                                   
    CODE_FILE=../src/network/networkd-network.c                                                                                                                                                
    CODE_FUNC=network_get                                                                                                                                                                      
    MESSAGE=enp0s20: found matching network '/etc/systemd/network/10-enp0s20.network'                                                                                                          
    CODE_LINE=497                                                                                                                                                                              
    _BOOT_ID=<redacted>
    _SYSTEMD_INVOCATION_ID=<redacted
    _SOURCE_REALTIME_TIMESTAMP=1539028545269022                                                                                                                                                
Mon 2018-10-08 21:55:45.269419 CEST [<redacted>]
    _MACHINE_ID=<redacted>
    _HOSTNAME=sd-123456                                                                                                                                                                        
    PRIORITY=7                                                                                                                                                                                 
    SYSLOG_FACILITY=3                                                                                                                                                                          
    _SELINUX_CONTEXT=unconfined                                                                                                                                                                
    _SYSTEMD_SLICE=system.slice                                                                                                                                                                
    _TRANSPORT=journal                                                                                                                                                                         
    SYSLOG_IDENTIFIER=systemd-networkd                                                                                                     
    _UID=100                                                                                                                               
    _GID=102                                                                                                                               
    _COMM=systemd-network                                                                                                                  
    _EXE=/lib/systemd/systemd-networkd                                                                                                                           
    _CMDLINE=/lib/systemd/systemd-networkd                                                                                                                       
    _CAP_EFFECTIVE=3c00                                                                                                                                         
    _SYSTEMD_CGROUP=/system.slice/systemd-networkd.service                                                                                              
    _SYSTEMD_UNIT=systemd-networkd.service                                                                                                           
    _PID=736                                                                                                                                           
    CODE_FILE=../src/libsystemd/sd-netlink/sd-netlink.c                                                                                                  
    CODE_FUNC=process_reply                                                                                                                                      
    ERRNO=22                                                                                                                                                     
    MESSAGE=sd-netlink: callback failed: Invalid argument                                                                                                           
    _BOOT_ID=<redacted>
    _SYSTEMD_INVOCATION_ID=<redacted>
    CODE_LINE=348                                                                                                                                             
    _SOURCE_REALTIME_TIMESTAMP=1539028545269419                                                                                                       
```

I don't know if settings those two options is enough, I did not try that.

Moreover, even if I might be able to re-install the system later on, this problem prevents network connectivity, the host is physically not accessible to me, in a different country, and I don't have remote console access (only a fairly limited rescue mode is available), so finding the minimal .network is beyond my resources. (And patience!)

Feel free to close the bug report if the above information is not sufficient, I can open a new one if there is a need or opportunity to do so later on.

**Comment by yuwata** [comment]
Thank you for the information. I will try to investigate this. When you will get more logs, then please provide them.

**Comment by yuwata** [comment]
Fix is waiting in #10201.
